Libraries/DeviceInfo/DeviceInfoModule.m

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "DeviceInfoModule.h"
+
+@interface DeviceInfoModule()
+
+@end
+
+@implementation DeviceInfoModule
+{
+
+}
+
+RCT_EXPORT_MODULE()
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+- (NSString*) deviceId
+{
+  struct utsname systemInfo;
+
+  uname(&systemInfo);
+
+  return [NSString stringWithCString:systemInfo.machine
+                                  encoding:NSUTF8StringEncoding];
+}
+
+- (NSString*) deviceName
+{
+  static NSDictionary* deviceNamesByCode = nil;
+
+  if (!deviceNamesByCode) {
+
+    deviceNamesByCode = @{@"i386"      :@"Simulator",
```

Comment:
I wasn't sure about this because there is no other way to get this information. If a new device pops up that it doesn't recognise it will fall back to the product family e.g. iPad.